cmd/cosign/cli/sign.go

```diff
@@ -172,11 +173,12 @@ func SignCmd(ctx context.Context, so SignOpts,
 		}
 		signer = sk
 	case so.KeyRef != "":
```

Comment:
Any reason to not just do the dupe detection on every upload? I think it's probably safe and cheap enough to do it on every upload, even though it wouldn't really be hit ever for keyless ones.

Reply:
Or maybe flip the polarity so the default is to detect dupes, but keyless mode skips that. (since I think we do want dupe detection for security keys, which are left out here).

Reply:
in order to detect dupes, we need a Verifier unless we can rely on deterministic signatures (which isn't standard practice). I can add support for security keys in this PR

> I think it's probably safe and cheap enough to do it on every upload, even though it wouldn't really be hit ever for keyless ones.

imho we shouldn't _require_ a dupe detector if it can be known that an upload should be performed. It'd just result in no-op boilerplate to avoid dereferencing nil

Reply:
> I can add support for security keys in this PR

done

Reply:
Ah yeah makes sense.